src/main/integrations/browser-window-session.ts

```diff
@@ -0,0 +1,97 @@
+import { Integration } from '@sentry/types';
+import { app, BrowserWindow } from 'electron';
+
+import { ELECTRON_MAJOR_VERSION } from '../electron-normalize';
+import { endSession, endSessionOnExit, startSession } from '../sessions';
+
+interface Options {
+  /**
+   * Number of seconds to wait before ending a session after the app loses focus.
+   *
+   * Default: 10 seconds
+   */
+  backgroundTimeoutSeconds?: number;
+}
+
+// The state can be, active, inactive, or waiting for a timeout
+type SessionState = 'active' | 'inactive' | { timer: NodeJS.Timeout };
+
+/**
+ * Tracks sessions as BrowserWindows focused.
+ *
+ * Supports Electron >= v12
+ */
+export class BrowserWindowSession implements Integration {
+  /** @inheritDoc */
+  public static id: string = 'BrowserWindowSession';
+
+  /** @inheritDoc */
+  public readonly name: string;
+
+  private _state: SessionState;
+
+  public constructor(private readonly _options: Options = {}) {
+    if (ELECTRON_MAJOR_VERSION < 12) {
+      throw new Error('BrowserWindowSession requires Electron >= v12');
+    }
+
+    this.name = BrowserWindowSession.id;
+    this._state = 'inactive';
+  }
+
+  /** @inheritDoc */
+  public setupOnce(): void {
+    app.on('browser-window-created', (_event, window) => {
+      window.on('focus', this._windowStateChanged);
+      window.on('blur', this._windowStateChanged);
+      window.on('show', this._windowStateChanged);
+      window.on('hide', this._windowStateChanged);
+
+      // when the window is closed we need to remove the listeners
+      window.once('closed', () => {
+        window.removeListener('focus', this._windowStateChanged);
+        window.removeListener('blur', this._windowStateChanged);
+        window.removeListener('show', this._windowStateChanged);
+        window.removeListener('hide', this._windowStateChanged);
+      });
+    });
+
+    // if the app exits while the session is active, end the session
+    endSessionOnExit();
+  }
+
+  private _windowStateChanged = (): void => {
+    // We need to test all windows for visibility AND focus
+    const aWindowIsActive = BrowserWindow.getAllWindows().some((window) => window.isVisible() && window.isFocused());
```

Comment:
Can we not use `BrowserWindow.getFocusedWindow()` and check for visibility?

```suggestion
    const windowIsActive = BrowserWindow.getAllWindows().some((window) => window.isVisible() && window.isFocused());
```

Reply:
I'll add a couple more tests and try it.

We want this to work with focus as well as visibility (hidden, minimised, etc)

I can't remember how or why but I came to this after quite a bit of testing.